doc/file-partition.md

```diff
@@ -0,0 +1,53 @@
+# File Partition:
+Since the blockchain is around ~140GB, storage of large files on an external drive is convenient.  If this is not done properly, the external drive will also contain high i/o-frequency LevelDB index files, protracting time for initial blockchain synchronization. This document describes how partition datadir files between the high-frequency/low-capacity "index" files and the low-frequency/high-capacity "blocks" files. Examples are given for macOS, but Linux / Windows should be similar. These instructions result in the following physical folder rearrangement:
+
+| Original Location       | New Location | Capacity Needs | i/o Frequency  |
+| ----------------------- | ------------ | -------------- | -------------- |
+| ${datadir}/blocks/index | ${datadir}   | low            | high           |
+| ${datadir}/blocks       | ${EXTERAL}   | high           | low            |
+| ${datadir}/chainstate   | n/a          | low            | high           |
+
```

Comment:
There seems to be some agreement that the last two rows in the above table are correct.  At issue still is whether the *first row* is correct or not -- it does reflect my experience, and is the impetus for this doc.  I identified this possible re-configuration pitfall (which I mistakenly made myself) via bitcoind file usage reported by the following command:
     lsof -p `cat ${datadir}/bitcoind.pid` | grep ldb\$
This motivated me to move the subordinate/child "index" folder back onto the internal drive and set up the soft links described here.  I did not "go the extra mile" though and verify i/o frequency demands of these index LeveDB files.  If someone could chime in and corroborate either way that would be appreciated.  Is there a blockchain indexing or synchronization test that we can run with the index files both *on* the external drive and *off-of* the external drive while capturing i/o frequency statistics as well as wall clock time?  Arguably this is currently "seat of my pants"; I first moved the entire blocks folder (including the index subfolder) to the external disk for internal capacity reasons, and it synched super super slowly.  Then I looked at open files as described above and spotted those "LevelDB" index folder files.  Moving them as described here sped things up to the point that performance seemed to match that with the default configuration of the blocks folder on the internal drive.  Basically I was just doing a simplistic `ls -lrt blk*.dat` every so often and watching how fast the blk*.dat files were growing in both cases.

A possible misperception here is that this doc is meant to be some sort of performance advice.  It is not, rather it's really just meant to be a note to help other developers/analysts who (like me) want to work on very low end commodity hardware yet perform initial synchronization quickly, without the large-capacity internal disk space requirements.

Neither are these meant as instructions to backup the blockchain for portability between bitcoin development instances either; as @sipa points out, the "blocks" folder 'needs to be kept in sync with the "index" folder', and the blocks folder is useless by itself for sure.  

Missing from this document is the (?reasonable?) expectation that bitcoind not be started until the external disk is mounted and likewise that the external disk not be dismounted/ejected until bitcoind is shut down.   To this point, I have not tried to see what happens, if after running in the configuration suggested here, the operator/developer accidentally tries to start bitcoin w/o the the external storage plugged in; I would hope that the index folder files are not corrupted if the blocks folder is not accessible.

This is merely intended to be more of a "pitfalls to avoid" type of document, rather than some sort of tuning advice.  I hope that this perspective is of use.